qiskit/ignis/experiments/calibration/instruction_data/instructions_definition.py

```diff
@@ -0,0 +1,353 @@
+                candidates = [_ for _ in candidates if _ <= cutoff_date]
+
+            candidates.sort(key=lambda x: x.date_time)
+
+            return candidates[-1].value
+        except KeyError:
+            raise QiskitError('No parameter value for %s and channel %s' % (name, ch.name))
+
+    def get_schedule(self, name: str, qubits: Tuple[int, ...],
+                     free_params: List[str] = None, group: Optional[str] = 'default') -> Schedule:
+        """
+        Args:
+            name: The name of the schedule to get.
+            qubits: The qubits for which to get the schedule.
+            free_params: The parameters that should remain unassigned.
+            group: The calibration group from which to draw the
+                parameters. If not specifies this defaults to the 'default' group.
+
+        Returns:
+            schedule: A deep copy of the template schedule with all parameters assigned
+                except for those specified by free_params.
+        """
+
+        # Get the schedule and deepcopy it to prevent binding from removing
+        # the parametric nature of the schedule.
+        sched = copy.deepcopy(self._schedules.get(name, Schedule()))
+
+        # Retrieve the channel indices based on the qubits and bind them.
+        binding_dict = {}
+        for ch in sched.channels:
+            if ch.is_parameterized():
+                index = self.get_channel_index(qubits, ch)
+                binding_dict[ch.index] = index
+
+        sched.assign_parameters(binding_dict)
+
+        # Loop through the remaining parameters in the schedule, get their values and bind.
+        if free_params is None:
+            free_params = []
+
+        binding_dict = {}
+        for inst in sched.instructions:
+            ch = inst[1].channel
+            for param in inst[1].parameters:
+                if param.name not in free_params:
+                    binding_dict[param] = self.parameter_value(param.name, ch, group=group)
+
+        sched.assign_parameters(binding_dict)
+
+        return sched
+
+    def get_circuit(self, name: str, qubits: Tuple, free_params: List[str] = None,
+                    group: Optional[str] = 'default', schedule: Schedule = None) -> QuantumCircuit:
+        """
+        Args:
+            name: The name of the gate to retrieve.
+            qubits: The qubits for which to generate the Gate.
+            free_params: Names of the parameters that will remain unassigned.
+            group: The calibration group from which to retrieve the calibrated values.
+                If unspecified this default to 'default'.
+            schedule: The schedule to add to the gate if the internally stored one is
+                not going to be used.
+
+        Returns:
+        """
+        if schedule is None:
+            schedule = self.get_schedule(name, qubits, free_params, group)
+
+        gate = Gate(name=name, num_qubits=len(qubits), params=list(schedule.parameters))
```

Comment:
I though we were going to aim for a different methodology when combining experiments?

Reply:
This can couple to the instruction map generation or scheduling (this gate can be used for not only calibration but also later algorithm execution with calibrated gate). Thus we need to specify the correct qubit number with the initial layout. I think this is a logical gate and no need to have exact mapping to the physical index but we can specify the physical qubits when adding a calibration entry (this is how other experimental class modules are implemented).

Reply:
Sounds good. I changed this with the latest pushes. The logic now uses `execute` and `initial_layout`.